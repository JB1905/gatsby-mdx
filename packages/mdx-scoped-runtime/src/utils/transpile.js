import transformMdx from '@mdx-js/mdx';
import remove from 'unist-util-remove';

export default ({ mdx }) => {
  if (!mdx) return { code: undefined };

  const code = transformMdx.sync(mdx, {
    skipExport: true,
    mdPlugins: [
      () => tree => {
        remove(tree, 'import');
      },
    ],
  });

  return { code };
};
