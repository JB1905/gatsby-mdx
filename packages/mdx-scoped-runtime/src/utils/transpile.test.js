import transpile from './transpile';

it('should transpile simple markdown', () => {
  expect(transpile({ mdx: '# Hi, world' }).code).toMatchSnapshot();
});

it('should drop imports', () => {
  expect(
    transpile({
      mdx: `import { List } from "./ui";

# Hey test
<List items={'howdy'.split('')} />
`,
    }).code
  ).toMatchSnapshot();
});

it('should handle layout', () => {
  expect(
    transpile({
      mdx: `import { List } from "./ui";
import Layout from "./Layout";

export default Layout

# Hey test
<List items={'howdy'.split('')} />
`,
    }).code
  ).toMatchSnapshot();
});
