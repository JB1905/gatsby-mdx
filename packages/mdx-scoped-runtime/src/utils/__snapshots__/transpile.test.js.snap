// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should drop imports 1`] = `
"

<MDXTag name=\\"wrapper\\"  components={components}><MDXTag name=\\"h1\\" components={components}>{\`Hey test\`}</MDXTag>
<List items={'howdy'.split('')} /></MDXTag>"
`;

exports[`should handle imported layout 1`] = `
"

<MDXTag name=\\"wrapper\\" Layout={Layout} layoutProps={props} components={components}>
<MDXTag name=\\"h1\\" components={components}>{\`Hey test\`}</MDXTag>
<List items={'howdy'.split('')} /></MDXTag>"
`;

exports[`should handle layout 1`] = `
"

<MDXTag name=\\"wrapper\\" Layout={({children}) => <div>{children}</div>} layoutProps={props} components={components}>
<MDXTag name=\\"h1\\" components={components}>{\`Hey test\`}</MDXTag>
<List items={'howdy'.split('')} />
<Clock /></MDXTag>"
`;

exports[`should transpile simple markdown 1`] = `
"

<MDXTag name=\\"wrapper\\"  components={components}><MDXTag name=\\"h1\\" components={components}>{\`Hi, world\`}</MDXTag></MDXTag>"
`;
